src/debug-generator/debug-generator.c

```diff
@@ -88,6 +165,12 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
                 return free_and_strdup_warn(&arg_default_unit, value);
 
+        } else if (streq(key, "systemd.break")) {
+
+                r = parse_breakpoint_from_string(value, &arg_breakpoints);
+                if (r < 0 && r != -EBADRQC)
+                        return log_warning_errno(r, "Failed to parse breakpoint value '%s': %m", value);
+
```

Comment:
as per early comment, i'd call parse_breakpoint_from_string() here with a temporary variable and on success then OR it into arg_breakpoints in a separate step.